From 5e81bbff8b295b1bd486f960ec0f2f77e927267d Mon Sep 17 00:00:00 2001
From: Igal Liberman <igall@marvell.com>
Date: Mon, 5 Dec 2016 13:10:58 +0200
Subject: [PATCH 1158/2241] fix: arm64: exception: ignore EL0 asynchronous
 external abort

There is an EL0 asynchronous external abort on some boot cases,
during file-system mount stage. The root cause of the abort is not
clear for now.

This patch adds a WA which allows to ignore EL0 asynchronous
external abort (in addition to EL1 asynchronous external abort
which is already ignored).

Change-Id: I317ddfc415b8a8c4ed7b087a50087fc2e7523b88
Signed-off-by: Igal Liberman <igall@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/34345
Tested-by: iSoC Platform CI <ykjenk@marvell.com>
Reviewed-by: Omri Itach <omrii@marvell.com>
---
 arch/arm64/kernel/entry.S | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/arch/arm64/kernel/entry.S b/arch/arm64/kernel/entry.S
index 698c4c6..db206c3 100644
--- a/arch/arm64/kernel/entry.S
+++ b/arch/arm64/kernel/entry.S
@@ -227,7 +227,14 @@ ENTRY(vectors)
 	ventry	el0_sync			// Synchronous 64-bit EL0
 	ventry	el0_irq				// IRQ 64-bit EL0
 	ventry	el0_fiq_invalid			// FIQ 64-bit EL0
+#ifdef CONFIG_IGNORE_ASYNC_ABORT
+	/* If CONFIG_IGNORE_ASYNC_ABORT is enabled, in a case of exeption,
+	 * ignore it be performing eret (in el0_ignore).
+	 */
+	ventry	el0_ignore
+#else
 	ventry	el0_error_invalid		// Error 64-bit EL0
+#endif
 
 #ifdef CONFIG_COMPAT
 	ventry	el0_sync_compat			// Synchronous 32-bit EL0
@@ -299,6 +306,10 @@ ENDPROC(el1_error_invalid)
 el1_ignore:
 	eret
 ENDPROC(el1_ignore)
+
+el0_ignore:
+	eret
+ENDPROC(el0_ignore)
 #endif
 
 /*
-- 
2.7.4

